### PR TITLE
WEB 803 - Add pagination for Sales History

### PR DESCRIPTION
**Pull request by joesobo**

### CODE REVIEW

**Comment by AndrewPark9517** [comment]
When user clicks on 'next page', the whole table goes into load. Could you make it so that the previous content remains, and only the newly fetched content is added. Probably means will need to have two separate isLoading variables. One for initial loading of table, and one for the next page tab.

**Comment by AndrewPark9517** [comment]
Also, hold off on this until we merge in WEB-810. We want to refactor the getSales requests into the client side as well.

File: components/SalesHistoryTable/index.tsx
Comment (joannerd) [inline components/SalesHistoryTable/index.tsx:126]: Needs another rebase. Sales history table is stuck in the loading state - we should call `setIsLoading(false)` in this `useEffect` to resolve that.

Let's also move this if statement inside of the `try` block.

**Review by cindyssun5 — CHANGES_REQUESTED** [review]

File: components/SalesHistoryTable/index.tsx
Comment (joannerd) [inline components/SalesHistoryTable/index.tsx:130]: Let's move `setIsLoading` to the end of the `try/catch` block so that if an error is thrown on line 129, `setIsLoading` is still invoked (otherwise, if an error is thrown, the table still stay loading).

File: components/SalesHistoryTable/index.tsx
Comment (cindyssun5) [inline components/SalesHistoryTable/index.tsx:136]: I believe we'll also need to setIsLoading(false) even if it goes into the error so that it won't be loading forever.

**Review by cindyssun5 — CHANGES_REQUESTED** [review]

**Review by cindyssun5 — APPROVED** [review]

**Pull request merged**
